andela/ah-backend-tesseract
# #159951969: Implement endpoints for comments on articles

## What does this PR do?

Have endpoints for comments (and replies to those comments) on articles working.

## Description of tasks to be completed?

Have the following endpoints working:
`POST /api/article/:slug/comments` for creating a comment
`GET /api/article/:slug/comments` for getting all comments on an article
`GET /api/article/:slug/comments/:id/replies` for getting all replies to a comment
`PUT /api/article/:slug/comments/:id` for updating a comment (or reply)
`DELETE /api/article/:comment/:id` for deleting a comment (or reply)

## How should this be manually tested?

- Check the README file for an example of the body for `post` and `put` requests.
- Make a request to any of the endpoints using the url of the review-app corresponding to this pull request
- Note that the `post`, `put` and `delete` endpoints require authentication. Authentication for the `get` endpoints is optional

## What are the relevant pivotal tracker stories?

#159951969
